ksql-parser/src/main/java/io/confluent/ksql/parser/rewrite/StatementRewriter.java

```diff
@@ -0,0 +1,1327 @@
+import io.confluent.ksql.parser.tree.IsNotNullPredicate;
+import io.confluent.ksql.parser.tree.IsNullPredicate;
+import io.confluent.ksql.parser.tree.Join;
+import io.confluent.ksql.parser.tree.KsqlWindowExpression;
+import io.confluent.ksql.parser.tree.LambdaExpression;
+import io.confluent.ksql.parser.tree.LikePredicate;
+import io.confluent.ksql.parser.tree.Literal;
+import io.confluent.ksql.parser.tree.LogicalBinaryExpression;
+import io.confluent.ksql.parser.tree.LongLiteral;
+import io.confluent.ksql.parser.tree.Node;
+import io.confluent.ksql.parser.tree.NotExpression;
+import io.confluent.ksql.parser.tree.NullIfExpression;
+import io.confluent.ksql.parser.tree.NullLiteral;
+import io.confluent.ksql.parser.tree.QualifiedNameReference;
+import io.confluent.ksql.parser.tree.Query;
+import io.confluent.ksql.parser.tree.QueryBody;
+import io.confluent.ksql.parser.tree.QuerySpecification;
+import io.confluent.ksql.parser.tree.Relation;
+import io.confluent.ksql.parser.tree.RenameColumn;
+import io.confluent.ksql.parser.tree.SampledRelation;
+import io.confluent.ksql.parser.tree.SearchedCaseExpression;
+import io.confluent.ksql.parser.tree.Select;
+import io.confluent.ksql.parser.tree.SelectItem;
+import io.confluent.ksql.parser.tree.SessionWindowExpression;
+import io.confluent.ksql.parser.tree.SetOperation;
+import io.confluent.ksql.parser.tree.SetSession;
+import io.confluent.ksql.parser.tree.ShowCatalogs;
+import io.confluent.ksql.parser.tree.ShowColumns;
+import io.confluent.ksql.parser.tree.ShowCreate;
+import io.confluent.ksql.parser.tree.ShowFunctions;
+import io.confluent.ksql.parser.tree.ShowPartitions;
+import io.confluent.ksql.parser.tree.SimpleCaseExpression;
+import io.confluent.ksql.parser.tree.SimpleGroupBy;
+import io.confluent.ksql.parser.tree.SingleColumn;
+import io.confluent.ksql.parser.tree.Statement;
+import io.confluent.ksql.parser.tree.Statements;
+import io.confluent.ksql.parser.tree.StringLiteral;
+import io.confluent.ksql.parser.tree.Struct;
+import io.confluent.ksql.parser.tree.SubqueryExpression;
+import io.confluent.ksql.parser.tree.SubscriptExpression;
+import io.confluent.ksql.parser.tree.SymbolReference;
+import io.confluent.ksql.parser.tree.Table;
+import io.confluent.ksql.parser.tree.TableElement;
+import io.confluent.ksql.parser.tree.TableSubquery;
+import io.confluent.ksql.parser.tree.TimeLiteral;
+import io.confluent.ksql.parser.tree.TimestampLiteral;
+import io.confluent.ksql.parser.tree.TumblingWindowExpression;
+import io.confluent.ksql.parser.tree.Values;
+import io.confluent.ksql.parser.tree.WhenClause;
+import io.confluent.ksql.parser.tree.Window;
+import io.confluent.ksql.parser.tree.WindowExpression;
+import io.confluent.ksql.parser.tree.WindowFrame;
+import io.confluent.ksql.parser.tree.WithQuery;
+import io.confluent.ksql.util.KsqlException;
+
+
+public class StatementRewriter extends DefaultAstVisitor<Node, Object> {
+
+  protected Node visitNode(final Node node, final Object context) {
```

Comment:
Isn't returning null the default behaviour of DefaultAstVisitor? So why override? 

This is also true of other methods. Would be good to remove those that can rely on the default implementation so that the functionality of this class shines through.

Reply:
As commented above, updated the implementation.

Reply:
Cool!